docs/getting_started/java/first_program_in_java/index.md

```diff
@@ -1,373 +1,757 @@
 
-It's time to start the Worker. A Worker is responsible for executing pieces of Workflow and Activity code.
+<!--SNIPSTART money-transfer-java-activity-interface {"selectedLines": ["3-19"]}-->
+<!--SNIPEND-->
 
-- It can only execute code that has been registered to it.
-- It knows which piece of code to execute from Tasks that it gets from the Task Queue.
-- It only listens to the Task Queue that it is registered to.
+#### Activity Implementation
 
-After The Worker executes code, it returns the results back to the Temporal server. Note that the Worker listens to the same Task Queue that the Workflow and Activity tasks are sent to. This is called "Task routing", and is a built-in mechanism for load balancing.
+This `AccountActivity` class implements three Activity methods: `withdraw`, `deposit`, and `refund`:
 
-<!--SNIPSTART money-transfer-project-template-java-worker-->
-[src/main/java/moneytransferapp/MoneyTransferWorker.java](https://github.com/temporalio/money-transfer-project-template-java/blob/4e1ca12d8d1c058c67d810efe8dcf68f7b318208/src/main/java/moneytransferapp/MoneyTransferWorker.java)
-```java
-public class MoneyTransferWorker {
+<!--SNIPSTART money-transfer-java-activity-implementation {"selectedLines": ["3-42"]}-->
+<!--SNIPEND-->
 
-    public static void main(String[] args) {
+* The `withdraw` method represents a process that calls a service to remove money from a source account.
+  It always succeeds.
+* The `deposit` process "attempts" to add money to a destination account.
+  Its success is controlled by the `activityShouldSucceed` Boolean, which is set to `true`.
+  If you change this to `false`, the "deposit" fails and throws an exception.
+* The `refund process mimics the `deposit` process.
 
-        // WorkflowServiceStubs is a gRPC stubs wrapper that talks to the local Docker instance of the Temporal server.
-        WorkflowServiceStubs service = WorkflowServiceStubs.newLocalServiceStubs();
-        WorkflowClient client = WorkflowClient.newInstance(service);
-        // Worker factory is used to create Workers that poll specific Task Queues.
-        WorkerFactory factory = WorkerFactory.newInstance(client);
-        Worker worker = factory.newWorker(Shared.MONEY_TRANSFER_TASK_QUEUE);
-        // This Worker hosts both Workflow and Activity implementations.
-        // Workflows are stateful so a type is needed to create instances.
-        worker.registerWorkflowImplementationTypes(MoneyTransferWorkflowImpl.class);
-        // Activities are stateless and thread safe so a shared instance is used.
-        worker.registerActivitiesImplementations(new AccountActivityImpl());
-        // Start listening to the Task Queue.
-        factory.start();
-    }
-}
-```
+:::note Recovering a failed Workflow
+
+It is not part of this tutorial, but you can try setting both `activityShouldSucceed` Booleans to false to see a Workflow Execution fail.
```

Comment:
Is this a Workflow Execution fail or a Workflow Task fail? If the Workflow in the Web UI has entered the FAILED state it is an Execution failure. If so, can you reset a Workflow that has entered the closed state? Does it spawn a new Workflow Execution? Once a Workflow is in a closed state it can't be re-opened, but I've never tried reseting one that is closed to see if it starts a new Execution. 
```suggestion
It is not part of this tutorial, but you can try setting both `activityShouldSucceed` Booleans to false to see a Workflow Task fail.
```

Reply:
Great catch.